zone/spell_effects.cpp

```diff
@@ -7290,6 +7281,22 @@ bool Mob::PassLimitClass(uint32 Classes_, uint16 Class_)
 	return false;
 }
 
+void Mob::DispelMagic(Mob* caster) 
+{
+	for (int slot = 0; slot < GetMaxTotalSlots(); slot++) {
+		if (
+			buffs[slot].spellid != SPELL_UNKNOWN &&
+			spells[buffs[slot].spellid].dispel_flag == 0 &&
+			!IsDiscipline(buffs[slot].spellid)
+		) {
+			if (caster) {
```

Comment:
Do we need the `TryDispel(caster->GetCasterLevel(spell_id), buffs[slot].casterlevel, effect_value)` that was here before?